lib/assertions.js

```diff
@@ -595,11 +595,16 @@ module.exports = function (expect) {
     expect.addAssertion('<function> to error [with] <any>', function (expect, subject, arg) {
         return expect(subject, 'to error').then(function (error) {
             expect.errorMode = 'nested';
-            if (error.isUnexpected && (typeof arg === 'string' || isRegExp(arg))) {
-                return expect(error, 'to have message', arg);
-            } else {
-                return expect(error, 'to satisfy', arg);
-            }
+            return expect.withError(function () {
+                if (error.isUnexpected && (typeof arg === 'string' || isRegExp(arg))) {
+                    return expect(error, 'to have message', arg);
+                } else {
+                    return expect(error, 'to satisfy', arg);
+                }
+            }, function (e) {
+                e._stack = error && error.stack;
+                throw e;
```

Comment:
Hmm, that already throws the original error, so the stack trace will presumably come out correct already: https://github.com/unexpectedjs/unexpected/blob/master/lib/Unexpected.js#L462-L464